homeassistant/components/airvisual_pro/config_flow.py

```diff
@@ -0,0 +1,66 @@
+"""Define a config flow manager for AirVisual Pro."""
+from __future__ import annotations
+
+from pyairvisual import NodeSamba
+from pyairvisual.node import NodeProError
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_IP_ADDRESS, CONF_PASSWORD
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import DOMAIN, LOGGER
+
+STEP_USER_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_IP_ADDRESS): str,
+        vol.Required(CONF_PASSWORD): str,
+    }
+)
+
+
+class AirVisualProFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle an AirVisual Pro config flow."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, str] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        if not user_input:
+            return self.async_show_form(step_id="user", data_schema=STEP_USER_SCHEMA)
+
+        await self.async_set_unique_id(user_input[CONF_IP_ADDRESS])
```

Comment:
Ip address is not an allowed unique id.

https://developers.home-assistant.io/docs/config_entries_config_flow_handler#unique-id-requirements

Reply:
https://github.com/home-assistant/core/pull/84902